mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -565,7 +566,10 @@ impl SwapWatcherLock {
     fn lock_taker(swap_ctx: Arc<SwapsContext>, fee_hash: Vec<u8>) -> Option<Self> {
         {
             let mut guard = swap_ctx.taker_swap_watchers.lock();
-            if !guard.insert(fee_hash.clone()) {
+            if guard
+                .insert_expirable(fee_hash.clone(), (), Duration::from_secs(TAKER_SWAP_ENTRY_TIMEOUT_SEC))
+                .is_some()
+            {
                 // There is the same hash already.
```

Comment:
not quite the same as the old logic.
the old insert didn't alter the expiration if the entry already existed. it just returned `false`:
```rust
    // Inserts new elements and removes any expired elements.
    //
    // If the key was not present this returns `true`. If the value was already present this
    // returns `false`.
    pub fn insert(&mut self, key: Key) -> bool {
        if let Entry::Vacant(entry) = self.0.entry(key) {
            entry.insert(());
            true
        } else {
            false
        }
    }
```

in contrast, the new change does reset the expiration time.

p.s. not that this new logic is incorrect (didn't check), but i assume that wasn't what u intended to do (hence no justification given)

Reply:
> not that this new logic is incorrect (didn't check)

I did check and it should not be a problem :)